### ISSUE TITLE
osu! crashed in a multiplayer room

### ISSUE THREAD

**Issue by Gonzalo-Bruna** [ppy/osu#12939]
Not sure if this is reported or not because I don't really know what caused the crash

**Describe the bug:**

I was in a multiplayer room and I downloaded and imported a beatmap, after the beatmap was imported the game crashed.

**osu!lazer version:** 

2021.523.0

**Logs:**

[runtime.log](https://github.com/ppy/osu/files/6533470/runtime.log)

I have other logs as well but I don't think they are needed?

**Comment by BojayB** [comment]
I had the same issue except a player selected a map and before I even downloaded it my game crashed. I was just sitting idle waiting to download the next map and when it popped up it crahsed. No clue what caused it just like this guy.

[database.log](https://github.com/ppy/osu/files/6679161/database.log)
[network.log](https://github.com/ppy/osu/files/6679162/network.log)
[performance.log](https://github.com/ppy/osu/files/6679163/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6679164/runtime.log)

![image](https://user-images.githubusercontent.com/41974762/122611975-49db4500-d050-11eb-8750-5842de52ff0f.png)

Even though I already uploaded the files, I don't want to open the game again until this is submitted just to make DAMN sure the logs don't change. All I know is I am on the latest version of osu as of writing. Just found out about the approach circle mod and have been using it since I downloaded the update.

**Comment by Susko3** [comment]
Looks like it's trying to load the approach different mod for mania. If you look at the gamemode toolbar, it looks like both osu! and mania are selected.

```
2021-06-18 20:10:49 [error]: An unhandled error has occurred.
2021-06-18 20:10:49 [error]: System.InvalidOperationException: There is no mod in the ruleset (mania) matching the acronym AD.
2021-06-18 20:10:49 [error]: at osu.Game.Online.API.APIMod.ToMod(Ruleset ruleset) in /var/lib/buildkite-agent/builds/debian-gnu-linux-vm-1/ppy/osu/osu.Game/Online/API/APIMod.cs:line 54
2021-06-18 20:10:49 [error]: at osu.Game.Screens.OnlinePlay.Multiplayer.Participants.ParticipantPanel.<>c__DisplayClass15_0.<OnRoomUpdated>b__1(APIMod m) in /var/lib/buildkite-agent/builds/debian-gnu-linux-vm-1/ppy/osu/osu.Game/Screens/OnlinePlay/Multiplayer/Participants/ParticipantPanel.cs:line 179
2021-06-18 20:10:49 [error]: at System.Linq.Enumerable.SelectArrayIterator`2.ToList()
2021-06-18 20:10:49 [error]: at osu.Game.Screens.OnlinePlay.Multiplayer.Participants.ParticipantPanel.<>c__DisplayClass15_0.<OnRoomUpdated>b__0() in /var/lib/buildkite-agent/builds/debian-gnu-linux-vm-1/ppy/osu/osu.Game/Screens/OnlinePlay/Multiplayer/Participants/ParticipantPanel.cs:line 179
```

**Comment by BojayB** [comment]
I had no idea that's what was going on. I didn't even know a mania map got loaded or how I was even able to select a mod to play in multiplayer. Thanks!

**Comment by peppy** [comment]
@smoogipoo I'm not sure if I assigned you to this issue after a discussion we had or not. Should I take a look into this instead? Looks like a duplicate of https://github.com/ppy/osu/issues/13567 as well.

**Comment by smoogipoo** [comment]
The issue in the OP is completely different from the one posted above.

You can look into it - this has been living in my inbox for the last little while.

**Comment by peppy** [comment]
Both hitting the same exception under similar conditions (beatmap change). I'll take a look.

**Comment by peppy** [comment]
Oh I see what you mean. This issue thread has two separate issues, only the second of which is the same one. I'll look into the newer one as this one is at a lower level.

**Comment by TheRealDust** [comment]
Similar Issue: Imported Skins, Beatmaps and Scores from stable (using the in-game menu), attempted to join a multiplayer room and the game crashed immediately after pressing "enter" on the (correct) password. Can't reproduce the issue.
[runtime.log](https://github.com/ppy/osu/files/8339578/runtime.log)
Osu version: 2022.319.0.0
OS: Windows 10

**Comment by peppy** [comment]
@TheRealDust in the future if unsure it's best to open a new issue. Your logs show an error happening in a different location to the one in this thread.

Relevant portion:

```csharp
2022-03-24 08:03:05 [error]: System.NullReferenceException: Object reference not set to an instance of an object.
2022-03-24 08:03:05 [error]: at osu.Game.Screens.OnlinePlay.Multiplayer.Participants.ParticipantPanel.OnRoomUpdated()
2022-03-24 08:03:05 [error]: at osu.Framework.Threading.ScheduledDelegate.RunTaskInternal()
2022-03-24 08:03:05 [error]: at osu.Framework.Threading.Scheduler.Update()
2022-03-24 08:03:05 [error]: at osu.Framework.Graphics.Drawable.UpdateSubTree()
```

**Comment by smoogipoo** [comment]
Should be resolved by #17431. Going to close this issue for now until it comes up again.
